### ISSUE TITLE
logstash -f <url> fails on https with useless error message

### ISSUE THREAD

**Issue by sgendler-stem** [elastic/logstash#8188]
- Version: docker.elastic.co/logstash/logstash:5.3.3
- Operating System: docker on OS X 10.12.6, but docker is based on Centos 7
- Config File (if you have sensitive info, please remove it): config doesn't matter, just host it on an https-enabled server (I'm using S3)
- Sample Data:
- Steps to Reproduce:

Create a pre-signed URL on S3, which always generates an https URL.  Pass that url to the docker image with a command such as the following:

docker run -it  docker.elastic.co/logstash/logstash:5.3.3 logstash -f 'https://...'

you'll get an error message like this:

ERROR logstash.agent - failed to fetch pipeline configuration {:message=>"Unable to fetch config from: https://...\nReason: End of file reached"}

Enabling debug logging wasn't any more useful.  There were no lower level log messages related to fetching the config file. It would be really useful to have debug output that provides the content it received - at one point, after I started fetching with http instead of https, my URL expired and S3 started serving up error messages (which also implies that the code wasn't checking the HTTP response code, since it wasn't a 2xx response).  That resulted in complaints about invalid config but it wasn't possible to see what it was.

tldr; - I just wasted about 2 hours trying to figure out how to get docker to load a config without including it in the image.  Some instruction on how to specify the url for the config file in a docker command would be nice.  I had to dig into your github repo to see how your docker image was constructed and what the entrypoint looked like in order to figure out what command to run and how to specify it.  Then I had to figure out that https doesn't work.  Then I had to figure out that status code's don't matter. Not exactly a user-friendly process.
